### DISCUSSION TITLE
How to generate a comment from macro argument

### DISCUSSION THREAD

**Discussion by vekexasia** [GoogleFeud/ts-macros#91] [Q&A] [answered]
Lets say i want to generate a comment

ex
```
function $hello(index:number) {
   // hey $index
}
```
how do i make it "output" `// hey 1` when using `$hello!(1)` ?

**Comment by volenslavchev** [comment, marked as answer]
Comments are always attached to a node, so you'll have to either:
- Also provide a node to the macro, the macro will expand to the node and add the comment above
- Create an empty statement and add the comment to it
- Find the node above the macro call (harder than it sounds) and append a trailing comment to it

[Playground](https://googlefeud.github.io/ts-macros?code=FAehAIEEBNoZ3AFwBYFNwGMD2BbHqA7RcAQwCMsA3dFdAraVUCAVzhoCcSBLAG24IBzAMK58RcADMWBDIm5YCSLOAAOLYt2IlitcIWjgskpGnD1GwabPmLwAEhKxReQogAU2V0QA05hqgAlOAA3sDgEeAcqIgsHEr29lwA7gCE7p6IAB5+XuKIAHIBfhZB4AC8AHyh4ZF10bHxmNkAdIhwLU7QAMoAnkRo8hgAMqhOAiJibu6lua3tLX1EJFkA0gLQLQCyLLzywwKoLvkAKhzclNwkuVNERYxtqFmIfogcLEG1EQC+gQDcwG+wGY4GE0R06BISlQOFUiF64DgiAh+VIBEMql4JAwNDMeTcpAo1HAWisMjkCgSXWObgAorD4Z5bohgmE6okUulMjlMMz7mUqjU6vUYnElHIsm0Ol0lrQhqNxkIaURuVKWpJsYgsBxei0MODEKh6XDet1kYb8u5AnNJQtZSt1ujtrt9odlYgzhcrjdvIUAo9nq93p86r8AUCyTZKQ5qczIETUEzfayvg4kiQ0hkJT78vzgoK2cKItgCEjMCReLwKs1bchuHAttiOFg9RXeACixFuCZ3KkMG3gg0xR3O7wYmoSNEJOVy5WWqpJ24SgF41RUMvGKvqABJdFPPxIlFuEdF5J1sfgdwLqcsoWdurdy8StX1s1HlXXtzBABk3-AqUPC0v0Rc0YQJGdPyIE9730Xh2BJHtVQWesACFeCwDAAGsr0XIhggAH3wmsXzgbosDiHEADE+ETSCWR-P9ALAvC7xg4USzLJj8gABSwBAINwxAWi4twOg2J4AHlJHcES8Ogtiux7WTEF4-jyhnABaABGQdRXieSFMfGTQJ4viKnUicb2EkzRJaMchBQcANPAHSoj08V5mlWBZUGbgRjGaAJndJCOg1ORtV1fUxkNY14TfICVWtYi7X6ZE1g2Z09m4A4CCOZlPUua5eV9fkAxeJBg0CVMFNg+DSi3dABKs5S4AAbWU1TwAAahcgBdaqFPqhNd0YLJqw6syeq0gy2LIcFMJm4UgRqujqzo+dBMW5bOzAcATjrBB6zctsEWQbFMN6VI9rME5ugq1B0COuAWFUVQ+NQQwtXAMhHrwDRyAvKFDF6ciAHJK3AM7iS+roBoiXaSHAAABdoNKeVRUDkNGOGbDgSSUEG4l5RhIdQaI4cstxrPfdpWqGtcRqeXrq2fBYZVS3z-MVSZfRmFcExtNV7XSp0diynK8t9ArvWK3N-UNQMKo+f4KaHfTUzDQFgUcZw4wTdIACIAAlUErLA-AAdW1XhoFSA2qo43RUDLGctIAJgAZgBHXoHdBrDZQ3p13AK2OBtu2HcUMtEZnAAWAFgB991DZOZ3EDtm5SywMc7KwQR3GN030L8ZJrdt+2VcT2NfViy6C-dQk11MdAYRNECaYjoA)
